agreement_sale_subscription_suspension/models/sale_subscription.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models
+
+
+class SaleSubscriptionSuspension(models.Model):
+    _inherit = 'sale.subscription'
+
+    def suspend_service_profile(self):
+        sp_ids = self.env['agreement.serviceprofile'].search([
+            ('agreement_id', '=', self.agreement_id.id)])
+        for sp_id in sp_ids:
+            sp_id.write({'state': 'suspended'})
+
+    def activate_service_profile(self):
+        sp_ids = self.env['agreement.serviceprofile'].search([
+            ('agreement_id', '=', self.agreement_id.id)])
+        for sp_id in sp_ids:
```

Comment:
```suggestion
        if sp_ids:
```

Reply:
IMO `if sp_ids:` is not needed as It won't iterate loop if there are no `sp_ids`